### PR TITLE
Allow to mutate PVCTemplate of Advanced StatefulSet & CloneSet

### PR DESCRIPTION
**Pull request by veophi**
Signed-off-by: mingzhou.swx <mingzhou.swx@alibaba-inc.com>

### Ⅰ. Describe what this PR does

Allow to mutate PVCTemplate of Advanced StatefulSet & CloneSet.

### CODE REVIEW

File: pkg/webhook/cloneset/validating/validation.go [hunk 1]
```diff
@@ -194,8 +194,9 @@ func (h *CloneSetCreateUpdateHandler) validateCloneSetUpdate(cloneSet, oldCloneS
 	clone.Spec.MinReadySeconds = oldCloneSet.Spec.MinReadySeconds
 	clone.Spec.Lifecycle = oldCloneSet.Spec.Lifecycle
 	clone.Spec.RevisionHistoryLimit = oldCloneSet.Spec.RevisionHistoryLimit
+	clone.Spec.VolumeClaimTemplates = oldCloneSet.Spec.VolumeClaimTemplates
 	if !apiequality.Semantic.DeepEqual(clone.Spec, oldCloneSet.Spec) {
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "updates to cloneset spec for fields other than 'replicas', 'template', 'lifecycle', 'scaleStrategy', 'updateStrategy', 'minReadySeconds' and 'revisionHistoryLimit' are forbidden"))
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "updates to cloneset spec for fields other than 'replicas', 'template', 'lifecycle', 'scaleStrategy', 'updateStrategy', 'minReadySeconds', 'volumeClaimTemplates' and 'revisionHistoryLimit' are forbidden"))
```
Comment (furykerry) [inline pkg/webhook/cloneset/validating/validation.go:199]: plz add ut for cloneset pvc template modification.

**Comment by willise** [comment]
@veophi I have a question that if controller should update controllerrevision to record the pvc template's change.

**Comment by veophi** [comment]
> 

These actions are already in our planning, but this pr only allow users modify the pvcTemplate manually.

**Comment by willise** [comment]
> @willise These actions are already in our planning, but this pr only allow users modify the pvcTemplate manually.

@veophi Is there any progress？

**Comment by HURUIZHE** [comment]
> > @willise These actions are already in our planning, but this pr only allow users modify the pvcTemplate manually.
> 
> @veophi Is there any progress？

+1. I need this feature. Now I can't rollback when I modified the cloneSet PVCTemplate

**Pull request merged**